py/examples/table_pagination_h2o3.py

```diff
@@ -0,0 +1,225 @@
+                        )
+                        for i in range(len(q.app.h2o_df.columns))
+                    ],
+                    rows=get_table_rows(q),
+                    resettable=True,
+                    downloadable=True,
+                    pagination=ui.table_pagination(
+                        total_rows=q.client.total_rows,
+                        rows_per_page=q.app.rows_per_page,
+                    ),
+                    events=[
+                        "page_change",
+                        "sort",
+                        "filter",
+                        "search",
+                        "reset",
+                        "download",
+                    ],
+                )
+            ],
+        )
+        q.client.initialized = True
+
+    # Check if user triggered any table action and save it to local state for allowing multiple
+    # actions to be performed on the data at the same time, e.g. sort the filtered data etc.
+    if q.events.h2o_table:
+        logger.info("table event occurred")
+
+        if q.events.h2o_table.page_change:
+            logger.info(f"table page change: {q.events.h2o_table.page_change}")
+            q.client.page_offset = q.events.h2o_table.page_change.get("offset", 0)
+
+        if q.events.h2o_table.sort:
+            logger.info(f"table sort: {q.events.h2o_table.sort}")
+            q.client.sort = q.events.h2o_table.sort
+            q.client.page_offset = 0
+
+        if q.events.h2o_table.filter:
+            logger.info(f"table filter: {q.events.h2o_table.filter}")
+            q.client.filters = q.events.h2o_table.filter
+            q.client.page_offset = 0
+
+        if q.events.h2o_table.search is not None:
+            logger.info(f"table search: {q.events.h2o_table.search}")
+            q.client.search = q.events.h2o_table.search
+            q.client.page_offset = 0
+
+        if q.events.h2o_table.download:
+            await download_h2o_table(q)
+
+        if q.events.h2o_table.reset:
+            logger.info("table reset")
+            q.client.search = None
+            q.client.sort = None
+            q.client.filters = None
+            q.client.page_offset = 0
+            q.client.total_rows = len(q.app.h2o_df)
+
+        # Update the rows on in our UI
+        # TODO: if you change where your table is located, this needs updating
+        q.page["table_card"].items[0].table.rows = get_table_rows(q)
+        q.page["table_card"].items[0].table.pagination.total_rows = q.client.total_rows
+
+    await q.page.save()
+
+
+def get_table_rows(q: Q):
+    logger.info(
+        f"Creating new table for rows: {q.client.page_offset} to {q.client.page_offset + q.app.rows_per_page}"
+    )
+
+    working_frame = prepare_h2o_data(q)
+
+    # Bring our limited UI rows locally to pandas to prepare for our ui.table
+    local_df = working_frame[
+        q.client.page_offset : q.client.page_offset + q.app.rows_per_page, :
+    ].as_data_frame()
```

Comment:
Why is conversion to pandas needed? Isn't it possible to iterate over `working_frame` and map it to `ui.table_row` directly?